pkg/objstore/s3/s3.go

```diff
@@ -173,8 +183,32 @@ func NewBucketWithConfig(logger log.Logger, config Config, component string) (*B
 	client.SetAppInfo(fmt.Sprintf("thanos-%s", component), fmt.Sprintf("%s (%s)", version.Version, runtime.Version()))
 
 	var sse encrypt.ServerSide
-	if config.SSEEncryption {
-		sse = encrypt.NewSSE()
+	if config.SSEConfig.Type != "" {
+		switch {
+		case config.SSEConfig.Type == "SSE-KMS":
+			sse, err = encrypt.NewSSEKMS(config.SSEConfig.KMSKeyID, config.SSEConfig.KMSEncryptionContext)
+			if err != nil {
+				return nil, errors.Wrap(err, "initialize s3 client SSE-KMS")
+			}
+
+		case config.SSEConfig.Type == "SSE-C":
+			key, err := ioutil.ReadFile(config.SSEConfig.EncryptionKey)
+			if err != nil {
+				return nil, err
+			}
+
+			sse, err = encrypt.NewSSEC(key)
+			if err != nil {
+				return nil, errors.Wrap(err, "initialize s3 client SSE-C")
+			}
+
+		case config.SSEConfig.Type == "SSE-S3":
+			sse = encrypt.NewSSE()
+
+		default:
+			sseErrMsg := errors.New("A type of SSE-S3, SSE-C, or SSE-KMS was not provided in sse_config")
```

Comment:
I think we just need `errors.Errorf` with one variable. Something like 

`errors.Errorf("not supported type %q was provided. Only ... suported") (: